ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -527,12 +527,66 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+/// Merges a `producer` of SignalProducers down into a single producer, biased toward the producers
+/// added earlier. Returns a SignalProducer that will forward signals from the original producers
+/// as they arrive.
+public func merge<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+
+	let relay = SignalProducer<T, E> { relayObserver, relayDisposable in
+		
+		let inFlight = Atomic(1)
```

Comment:
Open question: should this be implemented with `OSAtomicIncrement` and `OSAtomicDeccrement`?